zerver/lib/topic.py

```diff
@@ -82,6 +82,13 @@ def get_resolved_topic_condition_sa() -> ColumnElement[Boolean]:
     return resolved_topic_cond
 
 
+def get_followed_topic_condition_sa() -> ColumnElement[Boolean]:
+    followed_topic = UserTopic.objects.filter(visibility_policy=UserTopic.VisibilityPolicy.FOLLOWED)
+    followed_topic_names = [topic["topic_name"] for topic in followed_topic.values("topic_name")]
+    follow_topic_cond = column("subject", Text).in_(followed_topic_names)
+    return follow_topic_cond
```

Comment:
This query is wrong -- topics with the same name as a followed topic in another stream will incorrectly pass this check.

We also need to do the join in the database; fetching every topic in Python is going to be too expensive if a user has thousands of followed topics.

Reply:
I tested this and made two topics with same name "printer" one which I was following and another I was not. 
![image](https://github.com/zulip/zulip/assets/91622060/18f390e2-c4b1-4418-a39c-f714a9fa58b2)
And when I added a print statement for `followed_topic_names`  after line 87. It displayed "printer" only one time. This was the result of the print command.
```['old plotter is', 'bridge', "last WEB APP isn't skipping", '✔ API', 'last plotter is de-duping erratically', 'green URLS has been skipping erratically', "last server wasn't running", '✔ little nas has been slowly', 'first server running', 'printer', 'PLOTS', 'leased desktop has been', "red desktop wasn't linking"]```
Additionally the topic "printer" from stream `design` is not coming when filter is:followed is applied so I guess it's working fine as it is. 
![image](https://github.com/zulip/zulip/assets/91622060/5a0c1938-6591-4034-a814-346281145d23)